Detectors/AOD/src/AODProducerWorkflowSpec.cxx

```diff
@@ -1484,6 +1520,22 @@ void AODProducerWorkflowDPL::run(ProcessingContext& pc)
 
   cacheTriggers(recoData);
 
+  int collisionID = 0;
+  mIndexTableMFT.resize(recoData.getMFTTracks().size());
+  mIndexTableFwd.resize(recoData.getMCHTracks().size() * 3); // take an upperbound to the size of the FwdTrack table
+
+  auto& trackReffwd = primVer2TRefs.back();
+  fillIndexTablesPerCollision(trackReffwd, primVerGIs);
+  collisionID = 0;
+  for (auto& vertex : primVertices) {
+    auto& trackReffwd = primVer2TRefs[collisionID];
+    fillIndexTablesPerCollision(trackReffwd, primVerGIs); // this function must follow the same track order as 'fillTrackTablesPerCollision' to fill the map of track indices
+    collisionID++;
+  }
+
+  mGIDToTableFwdID.clear();
+  mGIDToTableMFTID.clear();
```

Comment:
Why are you resetting these maps here? They are used in the fillTrackTablesPerCollision which is called below, right? Should not it be e.g. here: https://github.com/AliceO2Group/AliceO2/blob/d7e376507aa3a4cc5efd7874aca27b48cf452851/Detectors/AOD/src/AODProducerWorkflowSpec.cxx#L1671
Also, I think you should set to 0 the mIndexFwdID and mIndexMFTID,

Reply:
Corrected, thanks. Also removed the filling of mGIDToTableFwdID and mGIDToTableMFTID in addToFwdTracksTable and addToMFTTracksTable, since they are already filled in fillIndexTablesPerCollision.